### DISCUSSION TITLE
New to using a database, need guidance/help

### DISCUSSION THREAD

**Discussion by JamesSingleton** [supabase/supabase#22026] [Questions]
I am going to start out by apologizing if this is the wrong place to ask for help, but I thought I would give it a shot here. I have a website for college football news and use Sanity for all of my content and Next.js as the framework. However, I am getting to a point where I need a DB for content that I want to add to the website.

I need to add a route that is protected that will allow people to log in and vote for their Top 25 teams each week of the season which would eventually show on the website like this:
![image](https://github.com/supabase/supabase/assets/21000200/f08205aa-ee36-4edb-a874-d008385798a4)

I also want to be able to break it down for which voter votes for which team, something like this:
![image](https://github.com/supabase/supabase/assets/21000200/aaa36008-3d01-4972-8ddf-e7b56d02256c)

I have started the process of creating a new project in Supabase and will be using Drizzle ORM and I am currently at the part of needing to create tables. This is where I am struggling as I don't want to mess up. I will be using Clerk for my auth, so I don't think I need to worry about that. However, I don't know what that table _should_ look like or if that is automatically created somehow. Anyways... this is what I have kind of come up with:

```ts
export const ballots = pgTable('ballots', {
  id: serial('id').primaryKey(),
  user_id: varchar('user_id', { length: 256 }),
  year: varchar('year', { length: 256 }),
  week: varchar('week', { length: 256 }),
  team_1: varchar('team_1', { length: 256 }),
  team_2: varchar('team_2', { length: 256 }),
  team_3: varchar('team_3', { length: 256 }),
  team_4: varchar('team_4', { length: 256 }),
  team_5: varchar('team_5', { length: 256 }),
  team_6: varchar('team_6', { length: 256 }),
  team_7: varchar('team_7', { length: 256 }),
  team_8: varchar('team_8', { length: 256 }),
  team_9: varchar('team_9', { length: 256 }),
  team_10: varchar('team_10', { length: 256 }),
  team_11: varchar('team_11', { length: 256 }),
  team_12: varchar('team_12', { length: 256 }),
  team_13: varchar('team_13', { length: 256 }),
  team_14: varchar('team_14', { length: 256 }),
  team_15: varchar('team_15', { length: 256 }),
  team_16: varchar('team_16', { length: 256 }),
  team_17: varchar('team_17', { length: 256 }),
  team_18: varchar('team_18', { length: 256 }),
  team_19: varchar('team_19', { length: 256 }),
  team_20: varchar('team_20', { length: 256 }),
  team_21: varchar('team_21', { length: 256 }),
  team_22: varchar('team_22', { length: 256 }),
  team_23: varchar('team_23', { length: 256 }),
  team_24: varchar('team_24', { length: 256 }),
  team_25: varchar('team_25', { length: 256 }),
})

export const rankings = pgTable('rankings', {
  id: serial('id').primaryKey(),
  year: varchar('year', { length: 256 }),
  week: varchar('week', { length: 256 }),
  team: varchar('team', { length: 256 }),
  votes: varchar('votes', { length: 256 }),
  rank: varchar('rank', { length: 256 }),
  user_id: varchar('user_id', { length: 256 }),
  ballot_id: varchar('ballot_id', { length: 256 }),
}, (t) => ({
  unq: unique().on(t.ballot_id)
}))
```

I will be pulling in the teams from my Sanity CMS and storing the ID for each team for that specific team ID, I _think_ that should work, but no 100%. Am I on the right track here? Am I missing something? One thing that I am kind of unsure about is adding the individual teams to the `votes` table. I want the team's name along with their logo (url from sanity) stored when they vote. However, I wasn't sure if creating a `teams` table would be appropriate as I have them stored in Sanity and feel like that would be duplicating the effort?

**Comment by GaryAustin1** [comment]
You don't need auth info in a public user table.  Supabase will handle logins/passwords etc.   EDIT this assume you planned to use Supabase auth and don't already have an auth system.

If you want some key info from that auth data you can use an insert trigger to copy out username, email and their user id to your public user table.  I personally link my other tables (like your votes table) to my public users table (actually called profiles).

Info on user management: https://supabase.com/docs/guides/auth/managing-user-data#using-triggers

Not commenting or looking at your general table structure.
